### PR TITLE
Kits branch

### PR DESCRIPTION
**Pull request by kit10ten**
Added a new encoding solution which incorporates SALTING.

### CODE REVIEW

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/Key.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            key[i] = ThreadLocalRandom.current().nextInt(1, 10);
+        }
+        this.key = key;
+    }
+
+    public int[] getKey() {
+        return key;
+    }
+    
+}
```
Comment (WhirrAlt) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/Key.java:20]: Wants a new line at end of file

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/MessageProcessor.java [hunk 2]
```diff
@@ -108,4 +136,23 @@ private char shiftPrintableChar(char ch, int shift) {
...
+
+        String saltedString = sb + input;
+
+        return saltedString;
+    }
+
+    public String removeFirstFiveChar(String s){
+        return s.substring(5);
+     }
+}
```
Comment (WhirrAlt) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/MessageProcessor.java:158]: Wants another line - remove comments throughout the code

**Review by WhirrAlt — COMMENTED** [review]
Clean-up files from comments and adding lines to end of the file

**Comment by WhirrAlt** [comment]
Tests work, maybe make the textbox clear when encode is clicked

**Review by GabrielParle — COMMENTED** [review]
looks good happy to merge

**Review by GabrielParle — APPROVED** [review]
im happy

**Review by Astra2707 — APPROVED** [review]
Looks ready to merge 👍

**Pull request merged**